core/thread/inc/ROOT/TReentrantRWLock.hxx

```diff
@@ -123,6 +124,67 @@ struct RecurseCounts {
    size_t GetLocalReadersCount(local_t &local) { return fReadersCount[local]; }
 
 
+};
+
+struct RecurseCountsLocalSpin {
+   using ReaderColl_t = std::unordered_map<std::thread::id, int>;
+   size_t fWriteRecurse; ///<! Number of re-entry in the lock by the same thread.
+
+   std::thread::id fWriterThread; ///<! Holder of the write lock
+   ReaderColl_t fReadersCount;    ///<! Set of reader thread ids
+
+   std::atomic_flag fSpinLock;    ///<! MSVC doesn't support = ATOMIC_FLAG_INIT;
+
+   using local_t = std::thread::id;
+
+   RecurseCountsLocalSpin() {
+      // MSVC doesn't support fSpinLock=ATOMIC_FLAG_INIT; in the class definition
+      std::atomic_flag_clear( &fSpinLock );
+   }
+
+   local_t GetLocal() { return std::this_thread::get_id(); }
+
+   void IncrementReadCount(local_t &local) { ++(GetLocalReadersCount(local)); }
+
+   template <typename MutexT>
+   void IncrementReadCount(local_t &local, MutexT &)
+   {
+      IncrementReadCount(local);
+   }
+
+   void DecrementReadCount(local_t &local) { --(GetLocalReadersCount(local)); }
+
+   template <typename MutexT>
+   void DecrementReadCount(local_t &local, MutexT &)
+   {
+      DecrementReadCount(local);
+   }
+
+   void ResetReadCount(local_t &local, int newvalue) {
+      GetLocalReadersCount(local) = newvalue;
+   }
+
+   bool IsNotCurrentWriter(local_t &local) { return fWriterThread != local; }
+
+   void SetIsWriter(local_t &local)
+   {
+      // if (fWriteRecurse == std::numeric_limits<decltype(fWriteRecurse)>::max()) {
+      //    ::Fatal("TRWSpinLock::WriteLock", "Too many recursions in TRWSpinLock!");
+      // }
```

Comment:
No commiting of dead code, please.

Reply:
I see this more as a temporarily hidden debugging code :)